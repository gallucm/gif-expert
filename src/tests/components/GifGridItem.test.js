const { shallow } = require("enzyme");
const { GifGridItem } = require("../../components/GifGridItem");

describe("Pruebas en <GifGridfItem />", () => {
    
    const title = "Titulo";
    const url = "https://localhost:/algo.jpeg";
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    const classDiv = 'animate__bounce';

    test("Se debe cargar el componente <GigGridItem/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un <p> con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe( title );
    });

    test('Debe tener un img con la url y el alt con el title', () => {
        const img = wrapper.find('img');
        const src = img.prop('src');
        const alt = img.prop('alt');

        expect(src).toBe(url);
        expect(alt).toBe(title);       
    });

    test('El div padre debe tener la clase "animate__bounce', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes(classDiv)).toBe(true);
    });        
    
});
