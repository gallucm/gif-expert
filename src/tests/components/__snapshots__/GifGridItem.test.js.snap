// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridfItem /> Se debe cargar el componente <GigGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Titulo"
    src="https://localhost:/algo.jpeg"
  />
  <p>
    Titulo
  </p>
</div>
`;
